src/frontend/hooks/goose-form.js

```diff
@@ -81,13 +81,25 @@ const useGooseForm = () => {
 
   const saveForm = useCallback(async () => {
     // TODO: Validation checks here.
-
     // Send data to server:
+
     try {
+      // files added 
       const files = images;
-      let imageUrls = [];
-
-      if (files.length > 0) {
+  
+      // all files uploaded 
+      let imageUrls2 = [];
+      
+      if (files.length > 0 || imageUrls.length > 0) {
+          // when no new files are added
+         if (files.length <= 0){
+          const imgArr = imageUrls.map((url) => url.imgDir)
+          imageUrls2 = imgArr
+          setImageUrls(imageUrls2)
+         }
+         
+        //  when new files are added, send them to api
+        if (files.length >= 1) {
```

Comment:
We can remove this if you alter the condition to only enter this if block if there are new images added